docs: add migration guide for bridge composition api (#1669)
nuxt/framework: docs/content/1.getting-started/4.bridge-composition-api.md

You can access injected helpers using `useNuxtApp`.

```diff
- import { useContext } from '@nuxtjs/composition-api`
+ import { useNuxtApp } from '#app'
+ const  { $axios } = useNuxtApp()
```

```vue
<script>
import { useNuxtApp } from '#app'
const { $axios } = useNuxtApp()
</script>
```

::alert{icon=👉}
`useNuxtApp()` also provides a key called `nuxt2Context` which contains all the same properties you would normally access from Nuxt 2 context, but it's advised _not_ to use this directly, as it won't exist in Nuxt 3. Instead, see if there is another way to access what you need. (If not, please raise a feature request or discussion.)
::

### `wrapProperty`

This helper function is not provided any more but you can replace it with the following code:

```js
const wrapProperty = (property, makeComputed = true) => () => {
  const vm = getCurrentInstance().proxy
  return makeComputed ? computed(() => vm[property]) : vm[property]
}
```

### `useAsync` and `useFetch`

There is currently no replacement for these two functions with bridge.

You can continue to use `useAsync` and `useFetch` by importing them from `@nuxtjs/composition-api`, which is shimmed by Nuxt Bridge.

### `useMeta`

> In order to use vue-meta, you can continue importing `useMeta` from `@nuxtjs/composition-api`, which is shimmed by Nuxt Bridge.

Author: ```suggestion
In order to use vue-meta, you can continue importing `useMeta` and `defineComponent` from `@nuxtjs/composition-api`, which is shimmed by Nuxt Bridge.

However, note that the existing limitations of `useMeta` continue, so it cannot be used in `<script setup>` and you must include a `head: {}` object within your `defineComponent`.

```js
import { defineComponent, useMeta } from '@nuxtjs/composition-api'

export default defineComponent({
  // You need to define an empty head to activate this functionality
  head: {},
  setup() {
    // This will allow you to set the title in head - but won't allow you to read its state outside of this component.
    const { title } = useMeta()
    title.value = 'My page'
  },
})
```
```

# Migrating to Bridge Composition API

> Nuxt Bridge provides access the Composition API syntax. It is specifically designed to be aligned with Nuxt 3. Because of this, there are a few extra steps to take when enabling Nuxt Bridge, if you have been using the Composition API previously.

Author: ```suggestion
Nuxt Bridge provides access to the Composition API syntax. It is specifically designed to be aligned with Nuxt 3. Because of this, there are a few extra steps to take when enabling Nuxt Bridge, if you have been using the Composition API previously.
```

## Using `@vue/composition-api`

> If you have been using just `@vue/composition-api` and not using the Nuxt community library, then things are very straightforward.

Reviewer: ```suggestion
If you have been using just `@vue/composition-api` and not `@nuxtjs/composition-api`, then things are very straightforward.
```

I guess it might be clear to state the package name before it has been mentioned. Or it might be better to flip these sections to talk about `@nuxtjs/composition-api` first

> 1. First, remove the plugin where you are manually registering the composition API. Nuxt Bridge will handle this for you.

Author: ```suggestion
1. First, remove the plugin where you are manually registering the Composition API. Nuxt Bridge will handle this for you.
```

   ```diff
   - import Vue from 'vue'
   - import VueCompositionApi from '@vue/composition-api'
   -
   - Vue.use(VueCompositionApi)
   ```

2. Otherwise, there is nothing you need to do. However, if you want, you can remove your explicit imports from `@vue/composition-api` and rely on Nuxt Bridge auto-importing them for you.

## Migrating from `@nuxtjs/composition-api`

> Nuxt Bridge implements the composition API slightly differently from `@nuxtjs/composition-api` and provides different composables (designed to be aligned with the composables that Nuxt 3 provides).

Author: ```suggestion
Nuxt Bridge implements the Composition API slightly differently from `@nuxtjs/composition-api` and provides different composables (designed to be aligned with the composables that Nuxt 3 provides).
```
